xsuite/src/proxy/proxy.ts

```diff
@@ -19,22 +19,28 @@ export class Proxy {
   proxyUrl: string;
   headers: HeadersInit;
   explorerUrl: string;
+  blockNonce: number | null;
```

Comment:
`undefined` seems more relevant than `null`, no? `null` means that there is no blockNonce at all, while here it is more that we don't know the blockNonce to use.

Reply:
Yes, I agree. But I think you could also see it like when the `blockNonce` is `null`, it means there is no `blockNonce` set, meaning the proxy has to not use any `blockNonce` in the request.

But the mean reason I put `null` is because when you set a `blockNonce` and then you want to clear it, it would make more sense to set it to `null` instead of `undefined`? I feel we should never explicitly assign `undefined` to a variable, but as you want.

Reply:
Let's go with null :+1: